crates/templates/src/app_info.rs

```diff
@@ -67,15 +72,61 @@ impl AppInfo {
         })
     }
 
+    fn from_existent_template(manifest_path: &Path) -> anyhow::Result<Self> {
+        // This has to be cruder, because (with the v2 style of component) a template
+        // is no longer valid TOML, so `from_existent_file` fails at manifest
+        // version inference.
+        let manifest_tpl_str = std::fs::read_to_string(manifest_path)?;
```

Comment:
Bit of a shame we have to read the whole file here, but the alternative is probably not worth it.